src/features/codeLensProvider.ts

```diff
@@ -28,12 +28,14 @@ class OmniSharpCodeLens extends vscode.CodeLens {
 export default class OmniSharpCodeLensProvider extends AbstractProvider implements vscode.CodeLensProvider {
 
     private _testManager: TestManager;
+    private _options: Options;
 
     constructor(server: OmniSharpServer, reporter: TelemetryReporter, testManager: TestManager)
     {
         super(server, reporter);
 
         this._testManager = testManager;
+        this._options = Options.Read();
```

Comment:
With this change, do we still turn codelens on/off as you switch between the prefernces tab and file?

Reply:
I don't think so. I would expect that this will require a restart of the extension to take effect.

Reply:
yeah indeed. Though I think this is consistent with the rest of the extension at the moment - we don't read any options in "real time" I think

Reply:
I think we're just inconsistent. For example, setting `csharp.suppressDotnetRestoreNotification` will take effect immediately. So will `csharp.disableCodeActions`. Maybe it's worth adding the extra logic to listen for `vscode.workspace.onDidChangeConfiguration`? Check out what we do in `CodeActionProvider` [here](https://github.com/OmniSharp/omnisharp-vscode/blob/2adadbd622dd5fe11035a380ef7c56dca1464562/src/features/codeActionProvider.ts#L16-L41)

Reply:
thanks for the tip - I did not see that before. I pushed the appropriate change. Indeed, the experience is much nicer now 👍